### DISCUSSION TITLE
How to stop deepgram microophone on call to action.

### DISCUSSION THREAD

**Discussion by Shridhar221B** [deepgram/community#839] [General help] [answered]
I am using deepgram microphone to record the audio but cannot stop it on action. Please help!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by M-C-Dev** [comment]
`microphone.mute()`

**Comment by SandraRodgers** [comment]
If you're using the Go SDK, see the documentation here: https://pkg.go.dev/github.com/deepgram/deepgram-go-sdk/pkg/audio/microphone#Microphone.Mute

If you are not using the Go SDK, could you give us more information about what language/SDK you are using and show us a code example?

**Reply by Shridhar221B** [reply]
I am using Python SDK. 

`
async def transcribe_audio_stream(websocket):
    """
    Async function to transcribe audio stream in real-time using Deepgram.
    """
    try:
        # Establish Deepgram connection for real-time transcription
        dg_connection = deepgram.listen.asynclive.v("1")

        # Event handlers for Deepgram connection events
        async def on_open(self, open, **kwargs):
            logging.info(f"Connection Open")

        async def on_message(self, result, **kwargs):
            global is_finals
            sentence = result.channel.alternatives[0].transcript

            if len(sentence) == 0:
                return

            if result.is_final:
                is_finals.append(sentence)
                if result.speech_final:
                    utterance = " ".join(is_finals)
                    is_finals = []
                    spk = result.channel.alternatives[0].words[0].speaker

                    # Send processed transcription using reusable WebSocket connection


        async def on_metadata(self, metadata, **kwargs):
            logging.info(f"Metadata: {metadata}")

        async def on_close(self, close, **kwargs):
            logging.info(f"Connection Closed")

        async def on_error(self, error, **kwargs):
            logging.info(f"Handled Error: {error}")

        dg_connection.on(LiveTranscriptionEvents.Open, on_open)
        dg_connection.on(LiveTranscriptionEvents.Transcript, on_message)
        dg_connection.on(LiveTranscriptionEvents.Metadata, on_metadata)
        dg_connection.on(LiveTranscriptionEvents.Close, on_close)
        dg_connection.on(LiveTranscriptionEvents.Error, on_error)

        options = LiveOptions(
            model="nova-2",
            language="en-US",
            smart_format=True,
            encoding="linear16",
            channels=1,
            sample_rate=16000,
            interim_results=True,
            utterance_end_ms=1000,
            endpointing=300,
            diarize=True
        )

        addons = {
            "no_delay": "True"
        }

        if await dg_connection.start(options, addons=addons) is False:
            logging.error("Failed to connect to Deepgram")
            return

        microphone = Microphone(dg_connection.send)
        microphone.start()

    except Exception as e:
        logging.error(f"Error transcribing audio: {e}")
        return await {"text": "Error transcribing audio",}
`

Right now you can see microphone get started with function call on call to the action but I want to set it off as well.

**Comment by SandraRodgers** [comment, marked as answer]
@Shridhar221B 

You can stop the microphone with `microphone.finish()` or you can mute it with `microphone.mute()`.

Here is example code which shows how you could do this with the Python SDK:

https://github.com/deepgram/deepgram-python-sdk/blob/02f6705ceb225fb8161fe3124a50b824c381414c/examples/advanced/streaming/mute-microphone/main.py
